Source/WebCore/editing/TextIterator.cpp

```diff
@@ -816,16 +812,15 @@ bool TextIterator::handleReplacedElement()
 
 static bool shouldEmitTabBeforeNode(Node& node)
 {
-    CheckedPtr renderer = node.renderer();
+    CheckedPtr cell = dynamicDowncast<RenderTableCell>(node.renderer());
```

Comment:
This is not identical to the previous check, which was doing:
```
bool isTableCell(const Node& node)
{
    auto* renderer = node.renderer();
    if (!renderer)
        return node.hasTagName(tdTag) || node.hasTagName(thTag);
    return renderer->isRenderTableCell();
}
```

In particular, you're no longer doing this part of the check:
```
    if (!renderer)
        return node.hasTagName(tdTag) || node.hasTagName(thTag);
```

Which seems like it could change behavior.

Reply:
That is a good catch and I hadn't appreciated that nuance. However, if there's no renderer it would always return false so I think what I did is still correct. In particular, this is the status quo:
```
if (!renderer || !isTableCell(node))
    return false;
```

Reply:
Ah, that's fair. I missed that. I think we're good then.